### PR TITLE
Cleanup unused code

### PR DESCRIPTION
**Pull request by tsubery**
The commit removes some unused code and simplifies other pieces.
It also solves a problem I encountered where the ```uid``` field wasn't a string like the reset of ueberauth strategies I used. Also changed the default scope according to how I understand Strava's documentation.

### CODE REVIEW

File: lib/ueberauth/strategy/strava.ex [hunk 1]
```diff
@@ -3,9 +3,7 @@ defmodule Ueberauth.Strategy.Strava do
   Strava Strategy for Überauth.
   """
 
-  use Ueberauth.Strategy, default_scope: "public",
-                          uid_field: :id,
-                          oauth2_module: Ueberauth.Strategy.Strava.OAuth
+  use Ueberauth.Strategy, default_scope: "view_private"
```
Comment (andrewhao) [inline lib/ueberauth/strategy/strava.ex]: I'd prefer the default_scope to be `public` still. I'm curious what your thought process was in making the default private?

**Comment by andrewhao** [comment]
Thank you for your contribution @tsubery! Quite honestly, I didn't have the forward thinking to implement a test suite for this, so I can't quite verify this works (yet). I assume you have this implementation running for your own project(s)?

**Comment by tsubery** [comment]
womm

**Review by andrewhao — APPROVED** [review]
Thanks, @tsubery. I'll work on a test suite when I get a chance 😜

**Pull request merged**